prov/gni/src/gnix_nameserver.c

```diff
@@ -95,23 +95,32 @@ static int gnixu_get_pe_from_ip(const char *ip_addr, uint32_t *gni_nic_addr)
                 if ((strstr(line,ip_addr) != NULL) && (strstr(line,"ipogif") != NULL)) {
 
                         ret = 0;
-                        scount = sscanf(line,"%s%s%s%s%s%s",dummy,dummy,dummy,mac_str,dummy,iface);
+                        scount = sscanf(line,"%s%s%s%s%s%s",fnd_ip_addr,dummy,
+					dummy,mac_str,dummy,iface);
                         if (scount != 6) {
                                 ret = -EIO;
                                 goto err;
                         }
 
-                        scount = sscanf(mac_str, "00:01:01:%02x:%02x:%02x", &w, &x, &y);
-                        if (scount != 3) {
-                                ret = -EIO;
-                                goto err;
-                        }
-
-                        /*
-                         * mysteries of XE/XC mac to nid mapping, see nid2mac in xt sysutils
-                         */
-
-                        *gni_nic_addr = (w << 16) | (x << 8) | y;
+			/*
+			 * check exact match of ip addr
+			 */
+			if(!strcmp(fnd_ip_addr,ip_addr)) {
+				fprintf(stderr,"found a match %s "
```

Comment:
Should this go through the log macros or is it a leftover debug statement?  Also, there was a similar case in GNI_INI in gnix_fabric.c that I had the same question about.